src/state/machines/helpers/turn.ts

```diff
@@ -0,0 +1,26 @@
+import { AnimalCard, GameState, PlayerState } from "@/state/types";
+import { concat, intersection } from "lodash";
+
+export const getAnimalBiomePairs = (player: PlayerState): AnimalCard[][] => {
+  const animalCards = player.hand.filter((card) => card.type === "animal") as AnimalCard[];
+  const animalCardBiomePairs = [];
+
+  for (let i = 0; i < animalCards.length; i++) {
+    for (let j = i + 1; j < animalCards.length; j++) {
+      if (intersection(animalCards[i].biomes, animalCards[j].biomes).length > 0) {
+        animalCardBiomePairs.push([animalCards[i], animalCards[j]]);
+      }
+    }
+  }
+
+  return animalCardBiomePairs;
+};
+
+export const checkAndAssignExtinctionTile = (gameState: GameState) => {
+  const player = gameState.players.find((player) => player.uid === gameState.turn.player)!;
+
+  if (player.hand.filter((card) => card.type === "disaster").length > 2 && gameState.extinctMarket.deck.length !== 0) {
+    gameState.extinctMarket.table = concat(gameState.extinctMarket.table, gameState.extinctMarket.deck[0]);
+    gameState.extinctMarket.deck = gameState.extinctMarket.deck.slice(1);
+  }
+};
```

Comment:
IMO this should be separated:
1) we have a guard the check if ext. tile should be given
2) we have a state to which we automatically transition where actual action is taken. Probable there will be a delaty for animations, multi-step state and so on..

Reply:
Yes this is temporary and will definitely be removed after working on the automatic events